test/test_worker_gem_independence.rb

```diff
@@ -19,22 +19,40 @@ def teardown
   end
 
   def test_changing_nio4r_version_during_phased_restart
-    change_gem_version_during_phased_restart old_app_dir: 'worker_gem_independence_test/old_nio4r',
+    change_gem_version_during_phased_restart server_opts: "--prune-bundler",
+                                             old_app_dir: 'worker_gem_independence_test/old_nio4r',
+                                             old_version: '2.7.1',
+                                             new_app_dir: 'worker_gem_independence_test/new_nio4r',
+                                             new_version: '2.7.2'
+  end
+
+  def test_changing_nio4r_version_during_phased_restart_with_preload_app
+    change_gem_version_during_phased_restart server_opts: "--preload",
+                                             old_app_dir: 'worker_gem_independence_test/old_nio4r',
+                                             old_version: '2.7.1',
+                                             new_app_dir: 'worker_gem_independence_test/new_nio4r',
+                                             new_version: '2.7.2'
+  end
+
+  def test_changing_nio4r_version_during_phased_restart_with_prune_bundler_and_preload_app
```

Comment:
Doesn't really make sense to have a setup like this but it doesn't do any harm either  🤷🏽‍♂️